src/cpp/benders/benders_core/include/BendersBase.h

```diff
@@ -227,6 +229,12 @@ class BendersBase {
                                PlainData::SubProblemData &subproblem_data,
                                const std::string &name,
                                const std::shared_ptr<SubproblemWorker> &worker);
+  // TODO to be rethink
+  std::shared_ptr<Outerloop::CriterionComputation> criterions_computation_;
```

Comment:
It seems weird to have a reference to `Outerloop` namespace within the `BendersBase` class....

Reply:
Yes I think that the criterion computation is a concept that is not necessarily linked to the outer loop as we can resort to it within Benders just to have outputs. However the use in Benders and in outer loop may have different objectives as in Benders it is just for outputs, whereas in outer loop, it is used to within the algo (to update the lambda), so I do not know if keeping a single concept is enough or if we have to split between something used to compute any outputs requested by the user and something to be used in the outer loop algorithm ? What do you think ?